### PR TITLE
forum create post get post worker

### PR DESCRIPTION
**Pull request by lbai001**
create forum post worker

auth0id -> auth0_id
function name changes in forum models
add .gitignore

cleans up console.log

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,9 @@ node_modules
 out
 typings
 seeds/Seeds.numbers
+models/listener-quiz
+seeds/
+log
+package-lock.json
+migrations/*_create_listener-quiz_*
+migrations/20171005003251_listener_quiz_auth0_id.js
```
Comment (RobAWilkinson) [inline .gitignore]: my same comment on these gww specific files applies here as well https://github.com/pushkin-consortium/api/pull/11/files#r145219935

File: worker.js [hunk 2]
```diff
@@ -231,15 +231,74 @@ function Worker() {
...
+                  .model('User')
+                  .forge()
+                  .save({
+                    created_at: new Date(),
+                    updated_at: new Date(),
+                    auth0_id: auth0_id
+                  })
+                  .then(user => {
+                    return user;
+                  });
```
Comment (RobAWilkinson) [inline worker.js]: `.then(data => { return data })` is unnecessary.

File: worker.js [hunk 3]
```diff
@@ -231,15 +231,74 @@ function Worker() {
...
+              }
+            });
+        }
+        return db
+          .model('User')
+          .forge()
+          .save()
+          .then(user => {
+            const profile = user.toJSON();
+            return profile;
```
Comment (RobAWilkinson) [inline worker.js]: `return user.toJSON()`

File: worker.js [hunk 4]
```diff
@@ -231,15 +231,74 @@ function Worker() {
...
+      this[`${quiz}.updateUserWithauth0_id`] = (uid, auth0_id) => {
+        return db
+          .model('User')
+          .where({ id: uid })
+          .set({ auth0_id: auth0_id })
+          .fetch()
+          .then(user => {
+            const profile = user.toJSON();
+            return profile;
+          });
```
Comment (RobAWilkinson) [inline worker.js]: `return user.toJSON()`

File: worker.js [hunk 5]
```diff
@@ -248,17 +307,36 @@ function Worker() {
...
+            //   .save()
+            //   .then(user => {
+            //     const json = data.toJSON();
+            //     const userInfo = user.toJSON();
+            //     console.log('user', userInfo);
+            //     return {
+            //       stimuli: json,
+            //       user: userInfo
+            //     };
+            //   });
```
Comment (RobAWilkinson) [inline worker.js]: please remove.

**Review by RobAWilkinson — DISMISSED** [review]
need to remove changes to gitignore to keep this open for pushkin users.

Also, now would be the best time to add in that [config package](https://www.npmjs.com/package/config) so these methods are not added to worker unless the user has specified they are using the forum functionality

File: .gitignore (hunk 1 above)
Comment (lbai001) [reply .gitignore]: check

File: worker.js (hunk 2 above)
Comment (lbai001) [reply worker.js]: check

File: worker.js (hunk 3 above)
Comment (lbai001) [reply worker.js]: check

File: worker.js (hunk 4 above)
Comment (lbai001) [reply worker.js]: check

File: worker.js (hunk 5 above)
Comment (lbai001) [reply worker.js]: check

**Pull request merged**